torch/lib/THC/generic/THCTensor.h

```diff
@@ -109,6 +109,7 @@ THC_API void THCTensor_(squeeze1d)(THCState *state, THCTensor *self, THCTensor *
 THC_API void THCTensor_(unsqueeze1d)(THCState *state, THCTensor *self, THCTensor *src, int dimension_);
 
 THC_API int THCTensor_(isContiguous)(THCState *state, const THCTensor *self);
+THC_API int THCTensor_(isContiguousFixOneSizedDimStride)(THCState *state, THCTensor *self);
```

Comment:
Uh can we please not add this function to the interface? Let's just keep it as a helper or as part of `newContiguous`, since that's the only place where it's used.